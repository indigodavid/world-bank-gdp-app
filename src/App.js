import { Route, Routes } from 'react-router-dom';
import './App.css';

function App() {
  return (
    <Routes>
      <Route path="/" />
    </Routes>
  );
}

export default App;
