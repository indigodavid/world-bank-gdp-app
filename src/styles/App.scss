*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

$color-bg: #121212;
$color-white: #ffffff;
$color-accent: #4369b2;

body {
  background-color: $color-bg;
  color: $color-white;
}