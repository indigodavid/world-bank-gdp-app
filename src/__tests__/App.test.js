import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import App from '../App';

const renderApp = () => {
  const { container } = render(
    <MemoryRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </MemoryRouter>,
  );
  return container;
};

describe('Integrations tests', () => {
  test('Match Snapshot', () => {
    expect(renderApp().firstChild).toMatchSnapshot();
  });

  test('Find Elements Header and Lazy loading elements', () => {
    renderApp();
    expect(screen.getByText(/World Bank/i)).toBeInTheDocument();
    setTimeout(() => {
      expect(screen.getByText(/East/i)).toBeInTheDocument();
    }, 1000);
  });
});
