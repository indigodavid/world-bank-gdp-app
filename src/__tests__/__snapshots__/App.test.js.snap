// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integrations tests Match Snapshot 1`] = `
<header>
  <div
    class="hamburgerIcon"
  >
    <button
      id="menuLink"
      type="button"
    >
      <svg
        class="activeIcon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
  <div
    class="navTitle"
  >
    ðŸŒŽ World Bank GDP data
  </div>
  <div
    class="searchBar"
  />
</header>
`;
